src/develop/pixelpipe_cache.c

```diff
@@ -130,17 +130,20 @@ uint64_t dt_dev_pixelpipe_cache_basichash(
   uint64_t hash = 5381;
 
   // we use the the imgid and pipe type for the hash
-  const uint32_t hashing_pipemode[2] = {(uint32_t)imgid, (uint32_t)pipe->type };
+  const uint32_t hashing_pipemode[3] = {(uint32_t)imgid,
+                                        (uint32_t)pipe->type,
+                                        (uint32_t)pipe->want_detail_mask };
 
   char *pstr = (char *)hashing_pipemode;
   for(size_t ip = 0; ip < sizeof(hashing_pipemode); ip++)
     hash = ((hash << 5) + hash) ^ pstr[ip];
 
-  // also use the details mask roi
+/*
```

Comment:
Maybe just remove this or add a comment about why it is commented out?

Reply:
I fully agree. Maybe merge now, I think it might be very good to explain exactly there why we take what parameters to calculate the hash and how we validate the integrity .
Will add this the next days.